extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/MethodNameParser.java

```diff
@@ -97,18 +97,23 @@ public Result parse(MethodInfo methodInfo) {
         int firstIndex = methodName.indexOf("First");
         int topIndex = methodName.indexOf("Top");
         if ((firstIndex != -1) || (topIndex != -1)) {
-            try {
-                String topCountStr = methodName.substring(Math.max(firstIndex, topIndex), byIndex)
-                        .replace("Top", "").replace("First", "");
-                if (topCountStr.isEmpty()) {
-                    topCount = 1;
-                } else {
-                    topCount = Integer.valueOf(topCountStr);
+            int maxFirstAndTopIndex = Math.max(firstIndex, topIndex);
```

Comment:
Hum, what about something like `findTop3ByFirstName`?
`firstIndex` would be 10 and `topIndex` would be 4 , resulting in `maxFirstAndTopIndex` 10.
This in turn would _not_ match `maxFirstAndTopIndex < byIndex` and so no `topCount` would be parsed.
Am I missing something?

Reply:
Good point. I'll make the checks more robust

Reply:
Should be better now